app/controllers/step_controller.rb

```diff
@@ -9,12 +9,18 @@ def current_tribunal_case
     @current_tribunal_case ||= TribunalCase.find_by_id(session[:tribunal_case_id])
   end
 
-  def update_and_advance(form_class, opts={})
+  def update_and_advance(attr, form_class, opts={})
     hash = params.fetch(form_class.name.underscore, {})
     @form_object = form_class.new(hash.merge(tribunal_case: current_tribunal_case))
     @next_step = params[:next_step].blank? ? nil : params[:next_step]
 
     if @form_object.save
+      # if we are reusing a step (e.g. income, income2, income3)
+      # we need to rename the 'income' attribute to 'income2' (or whatever) for the
+      # DecisionTree#destination call
+      # i.e. { income: "high" } -> { income2: "high" }
+      hash = opts[:as] ? { opts[:as] => hash[attr] } : hash
```

Comment:
I think I find it confusing owing to the variable name.  If I'm reading it correctly, 'hash' is name of the current step as expressed by the form name?  This can be overridden by passing in a `:as` option?